File: api-spring-boot/src/main/java/com/example/apispringboot/service/PrimeService.java
package com.example.apispringboot.service;

import org.springframework.stereotype.Service;

@Service
public class PrimeService {
    
    /**
     * Verifica se um número é primo
     */
    public boolean isPrime(int number) {
        if (number < 2) {
            return false;
        }
        
        for (int i = 2; i <= Math.sqrt(number); i++) {
            if (number % i == 0) {
                return false;
            }
        }
        
        return true;
    }
}

File: api-csharp/Services/PrimeService.cs
namespace ApiCsharp.Services
{
    public class PrimeService
    {
        /// <summary>
        /// Verifica se um número é primo
        /// </summary>
        /// <param name="number">Número a ser verificado</param>
        /// <returns>True se o número for primo, False caso contrário</returns>
        public bool IsPrime(int number)
        {
            if (number < 2)
            {
                return false;
            }

            for (int i = 2; i <= Math.Sqrt(number); i++)
            {
                if (number % i == 0)
                {
                    return false;
                }
            }

            return true;
        }
    }
}
